src/wagtailtrans/templatetags/wagtailtrans_tags.py

```diff
@@ -19,28 +21,32 @@ def _get_translations(page, homepage_fallback=True, include_self=True):
 
     """
     site = page.get_site()
+    available_translations = {}
     available_languages = get_languages_for_site(site)
-    if not include_self:
-        available_languages.remove(page.language)
-
-    page_translations = page.get_translations(
-        only_live=True, include_self=include_self)
-    available_translations = {
-        p.language.code: p for p in page_translations
-    }
+    if hasattr(page, 'language'):
+        if not include_self:
+            available_languages.remove(page.language)
+
+        page_translations = page.get_translations(
+            only_live=True, include_self=include_self)
+        available_translations = {
+            p.language.code: p for p in page_translations
+        }
 
     available_homepages = {}
+
     if homepage_fallback:
         available_homepages = {
             p.language.code: p
             for p in site.root_page.get_children().live().specific()
         }
 
-    translations = {}
+    translations = OrderedDict()
     for language in available_languages:
-        translation = available_translations.get(language.code)
-        if translation:
-            translations[language] = translation
+        if available_translations:
```

Comment:
We don't need the check here, you're doing a `get` anyway, which will return `None` if not found or dictionary is empty.